app/Image/TemporaryLocalFile.php

```diff
@@ -10,12 +10,23 @@
  */
 class TemporaryLocalFile extends NativeLocalFile
 {
-	public function __construct()
+	/**
+	 * @param string $fileExtension the file extension of the new temporary file incl. a preceding dot
+	 *
+	 * @throws \RuntimeException
+	 */
+	public function __construct(string $fileExtension)
 	{
-		$tempFilePath = tempnam(sys_get_temp_dir(), 'lychee');
-		if ($tempFilePath === false) {
-			throw new \RuntimeException('Could not create temporary file');
-		}
+		// We must not use the usual PHP method `tempnam`, because that
+		// method does not handle file extensions well, but our temporary
+		// files need a proper (and correct) extension for the MIME extractor
+		// to work.
+		$tempFilePath = sys_get_temp_dir() .
+			DIRECTORY_SEPARATOR .
+			'lychee-' .
+			strtr(base64_encode(random_bytes(12)), '+/', '-_') .
+			$fileExtension;
+		touch($tempFilePath);
```

Comment:
I'm confused by this code. Shouldn't it be in a loop so that if the generated random file name turns out to already exist, we generate a new one and try again? Instead we just a have `touch` here, which I don't think has the semantics of `O_CREAT | O_EXCL` that is typically required to make such temp file generation safe?

Reply:
You know more about that stuff than I do. I just thought, we have 96bit or randomness in the filename so collisions are negligible. But I agree, we should do it the right way. Need to read about `O_CREAT | O_EXCL` first.

Reply:
Fixed it in: https://github.com/LycheeOrg/Lychee/pull/1275/commits/74b9c45137908ab5918ee5c29259b14db5f79e45